infra/hooks/predeploy.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+echo "Retrieving cluster credentials"
+az aks get-credentials --resource-group ${AZURE_RESOURCE_GROUP} --name ${AZURE_AKS_CLUSTER_NAME}
+
+appConfigProviderStatus=$(helm status azureappconfiguration.kubernetesprovider -n azappconfig-system 2>&1)
+if [[ $appConfigProviderStatus == *"Error: release: not found"* ]]; then
+    echo "Azure App configuration Kubernetes provider not installed, installing"
+    helm install azureappconfiguration.kubernetesprovider oci://mcr.microsoft.com/azure-app-configuration/helmchart/kubernetes-provider --namespace azappconfig-system --create-namespace 
+else
+  echo "Azure App configuration Kubernetes provider installed, skipping"
```

Comment:
We are adding helm support for `azd` so this can be moved to be configured in `azure.yaml`. See https://github.com/Azure/azure-dev/pull/3196 for details.